Code/Framework/AzFramework/AzFramework/Input/Mappings/InputMappingAnd.cpp

```diff
@@ -8,9 +8,72 @@
 namespace AzFramework
 {
+    ////////////////////////////////////////////////////////////////////////////////////////////////
+    void InputMappingAnd::Config::Reflect(AZ::ReflectContext* context)
+    {
+        if (AZ::SerializeContext* serializeContext = azrtti_cast<AZ::SerializeContext*>(context))
+        {
+            serializeContext->Class<InputMappingAnd::Config, InputMapping::ConfigBase>()
+                ->Version(0)
+                ->Field("Source Input Channel Names", &Config::m_sourceInputChannelNames)
+            ;
+
+            if (AZ::EditContext* editContext = serializeContext->GetEditContext())
+            {
+                editContext->Class<InputMappingAnd::Config>("Input Mapping: And",
+                    "Maps multiple different input sources to a single output using 'AND' logic.")
+                    ->ClassElement(AZ::Edit::ClassElements::EditorData, "")
+                        ->Attribute(AZ::Edit::Attributes::AutoExpand, true)
+                        ->Attribute(AZ::Edit::Attributes::ChangeNotify, AZ::Edit::PropertyRefreshLevels::AttributesAndValues)
+                        ->Attribute(AZ::Edit::Attributes::NameLabelOverride, &InputMappingAnd::Config::GetNameLabelOverride)
+                    ->DataElement(AZ::Edit::UIHandlers::Default, &Config::m_sourceInputChannelNames, "Source Input Channel Names",
+                        "The source input channel names that will be mapped to the output input channel name.")
+                ;
+            }
+        }
+    }
+
+    ////////////////////////////////////////////////////////////////////////////////////////////////
+    AZStd::shared_ptr<InputMapping> InputMappingAnd::Config::CreateInputMapping(const InputContext& inputContext) const
+    {
+        if (m_outputInputChannelName.empty())
+        {
+            AZ_Warning("InputMappingAnd::Config", false, "Cannot create input mapping with empty name.");
+            return nullptr;
+        }
+
+        if (InputChannelRequests::FindInputChannel(InputChannelId(m_outputInputChannelName.c_str())))
+        {
+            AZ_Warning("InputMappingAnd::Config", false,
+                       "Cannot create input mapping '%s' with non-unique name.", m_outputInputChannelName.c_str());
+            return nullptr;
+        }
+
+        if (m_sourceInputChannelNames.empty())
+        {
+            AZ_Warning("InputMappingAnd::Config", false,
+                       "Cannot create input mapping '%s' with no source inputs.", m_outputInputChannelName.c_str());
+            return nullptr;
+        }
+
+        const InputChannelId outputInputChannelId(m_outputInputChannelName.c_str());
+        AZStd::shared_ptr<InputMappingAnd> inputMapping = AZStd::make_shared<InputMappingAnd>(outputInputChannelId,
+                                                                                              inputContext);
+        for (const InputChannelNameFilteredByDeviceType& soureInputChannelName : m_sourceInputChannelNames)
```

Comment:
Type in variable name `sourceInputChannelName`
```suggestion
        for (const InputChannelNameFilteredByDeviceType& sourceInputChannelName : m_sourceInputChannelNames)
```

Reply:
Good catch, fixed.